test/core_lang_auto/defaults.act

```diff
@@ -0,0 +1,54 @@
+def f(x, y='world'):
+    return x + ', ' + y
+
+actor main(env):
+    out1 = f('hello')
+    out2 = f('hello', 'you')
+    out3 = f('hello', y='there')
+    out4 = f(x='hello', y='there')
+    out5 = f(y='there', x='hello')
+    out6 = f('hello', None)
+    out7 = f(x='hello')
+    out8 = f(x='hello', y=None)
+    out9 = f(y=None, x='hello')
+
+#    print(out1)
+#    print(out2)
+#    print(out3)
+#    print(out4)
+#    print(out5)
+#    print(out6)
+#    print(out7)
+#    print(out8)
+#    print(out9)
+
+    if out1 != "hello, world":
```

Comment:
There's `testing.assertEqual()` which is quite a bit nicer from a coding perspective but it would introduce a dependency on the testing module which in turn relies on exceptions etc. Should we care about such dependencies? Maybe it does make sense to have some hierarchy of languages features so that core features only relies on core features and not stdlib etc etc?